server.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	availableSigners map[string]AvailableSigner
```

Comment:
This is the only big problem I have with this PR. Why does this need to be a global variable?

Can't you pass it as an argument to the server?

This should definitely be an argument to `ServerRun(host string, port int, availableSigners map[string]AvailableSigner)`.

You can create a `WaggleServer` struct and define the handlers on that struct if you want, so that they have access to the `availableSigners` member.

Reply:
We should avoid these kind of global variables. It limits how `waggle` can be used (e.g. you can't instantiate multiple waggle servers in the same process)`.

Reply:
Problem not in passing to `ServerRun` but in passing variable in each `mux`
```go
serveMux := http.NewServeMux()
serveMux.HandleFunc("/ping", pingRoute)
serveMux.HandleFunc("/sign/dropper", signDropperRoute)
```

Reply:
Not sure how I ll pass in `signDropperRoute`